accounts/views.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth import login, logout
+from django.contrib.auth.forms import AuthenticationForm
+from django.core.urlresolvers import reverse, reverse_lazy
+from django.http import HttpResponseRedirect
+from django.views import generic
+from django.views.generic import TemplateView
+
+from . import forms
+
+
+class LoginView(generic.FormView):
+    form_class = AuthenticationForm
+    success_url = reverse_lazy('accounts:home')
+    template_name = 'accounts/login.html'
+
+    def get_form(self, form_class=None):
+        if form_class is None:
+            form_class = self.get_form_class()
+        return form_class(self.request, **self.get_form_kwargs())
+
+    def form_valid(self, form):
+        login(self.request, form.get_user())
+        return super().form_valid(form)
+
+
+class LogoutView(generic.RedirectView):
+    url = reverse_lazy('home')
+
+    def get(self, request, *args, **kwargs):
+        logout(request)
+        return super().get(request, *args, **kwargs)
+
+
+class SignUpView(generic.CreateView):
+    form_class = forms.UserCreateForm
+    success_url = reverse_lazy('login')
+    template_name = 'accounts/signup.html'
+
+
+class Home(TemplateView):
+    template_name = 'accounts/home.html'
+
+    def get(self, request, *args, **kwargs):
```

Comment:
In this case it should absolutely be on `dispatch` because your authentication will actually be bypassed if the request uses a different verb. Also you should use the `login_required` decorator. Check out this article:

https://docs.djangoproject.com/en/dev/topics/class-based-views/intro/#decorating-the-class